crates/hir/src/lib.rs

```diff
@@ -3355,6 +3356,21 @@ impl Impl {
         let src = self.source(db)?;
         src.file_id.as_builtin_derive_attr_node(db.upcast())
     }
+
+    pub fn filter_self_ty_params(
```

Comment:
The name isnt really descriptive, I'm not sure I understand what this is filtering

Reply:
It filters an `impl`'s generic arguments pertaining to `self_ty`, e.g. for `impl<A, B, C> Trait<A, B> for Foo<C, B>` it will return `C` and `B` only. I failed to find a better name, do you have any suggestions? May be I could just add a comment with this example?

Reply:
Ye a, comment would be good for now